sphericart-torch/python/tests/test_autograd.py

```diff
@@ -18,37 +18,37 @@ def test_autograd_cartesian(xyz):
         l_max=4, backward_second_derivatives=True
     )
 
-    def compute(xyz):
-        sph = calculator.compute(xyz=xyz)
+    def compute(xyz_in):
+        sph = calculator.compute(xyz=xyz_in)
         assert torch.linalg.norm(sph) != 0.0
         return sph
 
     assert torch.autograd.gradcheck(compute, xyz, fast_mode=True)
     assert torch.autograd.gradgradcheck(compute, xyz, fast_mode=True)
 
     if torch.cuda.is_available():
-        xyz = xyz.to(device="cuda")
-        assert torch.autograd.gradcheck(compute, xyz, fast_mode=True)
-        assert torch.autograd.gradgradcheck(compute, xyz, fast_mode=True)
+        xyz_cuda = xyz.detach().cuda().requires_grad_(True)
```

Comment:
```suggestion
        xyz_cuda = xyz_in.detach().cuda().requires_grad_(True)
```

right?

Reply:
no, xyz_in is a parameter of the nested function compute. autograd.gradcheck takes that function in as a parameter and we pass in xyz_cuda as a variable.

Reply:
you're right, I got confused here